src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt

```diff
@@ -369,25 +365,36 @@ private fun HearingDto.toNomisUpdateHearing(): UpdateHearingRequest = UpdateHear
   internalLocationId = this.locationId,
 )
 
-private fun HearingOutcomeDto.toNomisCreateHearingResult(
-  adjudicationStatus: String,
-  hearingType: HearingDto.OicHearingType,
-): CreateHearingResultRequest =
-  CreateHearingResultRequest(
-    pleaFindingCode = this.plea!!.name,
-    findingCode = toNomisFindingCode(adjudicationStatus),
-    adjudicatorUsername = getAdjudicatorUsernameForInternalHearingOnly(hearingType),
+// logic for determining the NOMIS findingCode:
+// If separate outcome block exists use code, else use code from hearing.outcome
+private fun OutcomeHistoryDto.toNomisCreateHearingResult(): CreateHearingResultRequest {
+  val separateOutcome: OutcomeDto? = this.outcome?.outcome
+  val hearing = this.hearing!!
+  val findingCode = separateOutcome?.code?.name ?: hearing.outcome!!.code.name
+
+  return CreateHearingResultRequest(
+    pleaFindingCode = hearing.outcome!!.plea?.name ?: HearingOutcomeDto.Plea.NOT_ASKED.name,
+    findingCode = toNomisFindingCode(findingCode),
+    adjudicatorUsername = getAdjudicatorUsernameForInternalHearingOnly(
+      hearing.oicHearingType.name,
+      hearing.outcome.adjudicator,
+    ),
   )
+}
 
-private fun toNomisFindingCode(adjudicationStatus: String) = when (adjudicationStatus) {
+private fun toNomisFindingCode(outcome: String) = when (outcome) {
   "CHARGE_PROVED" -> "PROVED"
-  else -> adjudicationStatus
+  "ADJOURN" -> "S"
+  "DISMISSED" -> "D" // TODO confirm they still want to map to D
+  "REFER_POLICE" -> "REF_POLICE"
+  "REFER_INAD" -> "REF_INAD" // TODO confirm behaviour with john
```

Comment:
will require work in noms-api to ignore the creation of an outcome for REF_INAD and possibly just update a comment. Behaviour to be confirmed